docs/porting/standard_pin_names/pin_names_porting.md

```diff
@@ -0,0 +1,234 @@
+# Standard Pin Names Porting Guide
+
+This guide aims to help vendors to comply with the new pin naming guidelines.
+
+There is a set of generic guidelines that apply to all boards, and further sets of guidelines that additionally apply to boards that have certain compatible connectors. The current guideline documents can be found here:
+* [General Pin Names Design Document](https://github.com/ARMmbed/mbed-os/blob/feature-std-pin-name/docs/design-documents/hal/0004-pin-names-general-guidelines.md)
+* [Arduino Uno Pin Names Design Document](https://github.com/ARMmbed/mbed-os/blob/feature-std-pin-name/docs/design-documents/hal/0005-pin-names-Arduino-Uno-standard.md)
+
+## Generic Pin Names
+
+The generic guidelines currently define rules for the naming of LED pins, button pins and UART pins. In summary, the rules are:
+* LED pins are defined as `LEDx` (e.g. `LED0`, `LED1`)
+* Button pins are defined as `BUTTONx` (e.g. `BUTTON0`, `BUTTON1`)
+* UART pins for communication with host are defined as `USBTX` and `USBRX`
+* Pin aliases are allowed (e.g. `#define RED_LED LED1`)
+* Pin definitions must be a `#define`, not an `enum`
```

Comment:
Maybe not clear ?
    PA_0  = 0x00,
can be seen as pin definition ?

Reply:
@jeromecoutant can you clarify what you mean?

Reply:
Just a minor comment to say that BUTTON1 is a "pin", PA_0 also...
So all "pin" definitions can't be a `#define`, but maybe only "pin alias" definition ?

Reply:
I agree, although I suggest we keep specif pins (e.g. PA_0) out of the rules, as this changes from MCU vendor to other.
We already provide an example of defining this below in the guide :)